prod_chemical_release/scripts/combine_stewi_files.R

```diff
@@ -1,89 +1,216 @@
+#'@description A function to make a connection to the database
+#'@param con_type Whether to connect to postgres, mysql, or sqlite version
+#'@import DBI RMySQL RSQLite
+#'@return Database connection pointer object
+connect_to_db <- function(con_type){
+  switch(con_type,
+         "postgres" = dbConnect(RPostgreSQL::PostgreSQL(), 
+                                user = Sys.getenv("postgres_user"), 
+                                password = Sys.getenv("postgres_pass"), #
+                                host = Sys.getenv("postgres_host"), #
+                                dbname = db_name),
+         "mysql" = dbConnect(RMySQL::MySQL(), #Connect to database with .Renviron parameters
+                             username = Sys.getenv("mysql_user"), 
+                             password = Sys.getenv("mysql_pass"),
+                             host = Sys.getenv("mysql_host"), 
+                             port = 3306,
+                             dbname = db_name),
+         'sqlite' = dbConnect(RSQLite::SQLite(), "prod_chemical_release.sqlite")
+  ) %>% return()
+}
+
+#'@description A helper function to query database and receive the results. 
+#'Handles errors/warnings with tryCatch.
+#'@param query A SQL query string to query the database with
+#'@param con_type Whether to connect to postgres, mysql, or sqlite version
+#'@param schema The schema name to use if using a postgresql connection
+#'@import DBI dplyr
+#'@return Dataframe of database query results
+query_db <- function(query=NULL, con_type, schema){
+  if(is.null(query)) return(message("Must provide a query to send"))
+  con = connect_to_db(con_type)
+  query_result = tryCatch({
+    if(con_type == "postgres"){#Add schema tag
+      return(dbGetQuery(con, query %>% gsub("FROM ", paste0("FROM ",schema,"."), .)))
+    } else {
+      return(dbGetQuery(con, query))
+    }
+  },
+  error=function(cond){ message("Error message: ", cond); return(NULL) },
+  finally={ dbDisconnect(con) })
+  return(query_result)
+}
 
-#'@description A function to create a connection to the prod_factotum database using .Renviron parameters
+#'@description A helper function to send a statement to the database to update table entries. 
+#'Handles errors/warnings with tryCatch.
+#'@param statement A SQL query string to send to the database
+#'@param con_type Whether to connect to postgres, mysql, or sqlite version
+#'@param schema The schema name to use if using a postgresql connection
+#'@import DBI dplyr magrittr
+#'@return None. A SQL statement is passed to the database to make changes to the database.
+send_statement_db <- function(statement=NULL, con_type=NULL, schema){
+  if(is.null(statement)) return(message("Must provide a statement to send"))
+  con = connect_to_db(con_type=con_type)
+  tryCatch({
+    if(con_type == "postgres"){#Add schema tag
+      return(dbSendStatement(con, statement %>% 
+                               gsub("FROM ", "FROM ",schema,".", .) %>%

```

Comment:
Just catching this now, this should actually be:

gsub("FROM ", paste0("FROM ",schema,"."), .)

Attempts at eventually providing support to a flexible PostgreSQL adaptation with minimal query changes.